.github/workflows/release.yml

```diff
@@ -0,0 +1,76 @@
+name: Publish a Release
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version without leading "v" (1.0, 2.3.4, 0.1.0-pre1)'
+        required: true
+        default: ''
+      image_tags:
+        description: 'Extra docker image tags ("latest,test")'
+        required: true
+        default: 'latest,release'
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    permissions:
+      contents: write
+      packages: write
+    steps:
+    - name: Ensure version does not start with 'v'
+      uses: actions/github-script@v7
+      with:
+        github-token: ${{ github.token }}
+        script: |
+          version = context.payload.inputs.version;
+          if (/^v/.test(version)) throw new Error("Bad version number: " + version)
+    - uses: actions/checkout@v4
+    - name: Cache data/sources
+      uses: ./.github/cache-sources-action
+    - uses: actions/setup-java@v4
+      with:
+        java-version: '21'
+        distribution: 'temurin'
+        cache: 'maven'
+        server-id: ossrh
+        server-username: MAVEN_USERNAME
+        server-password: MAVEN_PASSWORD
+
+    - name: Check tag does not exist yet
+      run: if git rev-list "v${{ github.event.inputs.version }}"; then echo "Tag already exists. Aborting the release process."; exit 1; fi
+
+    - run: ./scripts/set-versions.sh "${{ github.event.inputs.version }}"
+    - run: ./scripts/build-release.sh
+    - run: ./scripts/test-release.sh "${{ github.event.inputs.version }}"
+    - name: Create tag
+      uses: actions/github-script@v7
+      with:
+        github-token: ${{ github.token }}
+        script: |
+          github.rest.git.createRef({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            ref: "refs/tags/v${{ github.event.inputs.version }}",
+            sha: context.sha
+          })
+    - run: mv target/*with-deps.jar planetiler-openmaptiles.jar
+    - run: sha256sum planetiler-openmaptiles.jar > planetiler-openmaptiles.jar.sha256
+    - run: md5sum planetiler-openmaptiles.jar > planetiler-openmaptiles.jar.md5
+    - name: Create Release
+      uses: softprops/action-gh-release@v1
+      with:
+        fail_on_unmatched_files: true
+        tag_name: v${{ github.event.inputs.version }}
+        draft: true
+        files: |
+          planetiler-openmaptiles.jar*
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - run: ./scripts/push-release.sh ${{ github.event.inputs.version }}
+      env:
+        GITHUB_ACTOR: ${{ github.actor }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        IMAGE_TAGS: ${{ github.event.inputs.image_tags }}
+        MAVEN_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+        MAVEN_PASSWORD: ${{ secrets.OSSRH_PASSWORD }}
```

Comment:
in the planetiler repo push-release.sh uploads to github package registry and maven central. I think all we care about here is uploading to dockerhub, so you could replace this step with the same thing that snapshot workflow does:

https://github.com/openmaptiles/planetiler-openmaptiles/blob/461ea675a18691a928742ab4e9ef07be68513f32/.github/workflows/snapshot.yml#L20-L26

Reply:
Updated. And I've added also 'Upload artifact'.